src/thread.cpp

```diff
@@ -47,6 +48,11 @@ POSSIBILITY OF SUCH DAMAGE.
 namespace libtorrent
 {
 
+	void this_thread::yield()
+	{
+		boost::this_thread::yield();
```

Comment:
Yes. I thought it's better there since I don't have to add the header to files that use it. It's also easier to change thread implementations. If you do something like this you may also want to change `libtorrent::thread` to use Boost.Thread to ensure that the thread_specific_ptrs are freed when the thread exits on Windows. If all the threads are long lived it doesn't matter that much though.

Reply:
master uses `std::thread` so the equivalent here would be `std::this_thread::yield()`